deluge/component.py

```diff
@@ -192,6 +192,8 @@ def on_pause(result):
             if self._component_timer and self._component_timer.running:
                 d = maybeDeferred(self._component_timer.stop)
                 d.addCallback(on_pause)
+                if hasattr(self, 'pause'):
```

Comment:
I think this should be similarly to how _component_stop() is done. In other words, we should call pause() and then add on_pause() as a callback which does a _component_timer.stop().

This way it's consistent with the other behaviour and the pause() resume() signatures do not need to include a result from a previous deferred.